### ISSUE TITLE
Null operator expected error at compilation but got it at runtime.

### ISSUE THREAD

**Issue by Bartmax** [dotnet/roslyn#1246]
```bool IsBool = myClass?.myInt == 4;```

this does not throw a compiler error, but we can get a runtime one if myClass is null because bool cannot be null.

I expected this to error at compilation. 

Posible suggestions for refactor should be either

change type to bool?
```bool? IsBool = myClass?.myInt == 4;```
assign default value on null something like :
```bool IsBool = myClass?.myInt ?? default(int) == 4;```
or
```bool IsBool = (myClass?.myInt ?? == 4) ?? default(bool);```

**Comment by gafter** [comment]
Can you please provide a complete program that illustrates the issue?

**Comment by gafter** [comment]
As described above, there should be no null reference exception in your code, and that is by design. You have not provided a test case that demonstrates an exception at runtime.

If I have misunderstood your issue, or you believe this resolution is not appropriate, please reopen this issue.

**Comment by Bartmax** [comment]
**TL;DR**
Roslyn does work as expected. My bad thinking it was a compiler error.
I'm providing details of what I were facing for completeness. 


When trying to reproduce the problem (in a simple scenario). You said, that should **not** throw at run-time (and it doesn't) but was exactly what were happening, so further investigation needed :

I was working with Open XML SDK and some code looks like this :

```
 if (runProperties.Bold != null)
    part.IsBold = runProperties.Bold.Val;
```

as I see this code, I try to change it to :
```
    part.IsBold = runProperties.Bold?.Val;
```

sound it should work, right ? well, it doesn't. Null reference exception.

 After further investigation, Val is not a bool but an `OnOffValue` class.
So the exception is throw in the implementation of `public static implicit operator bool(OnOffValue xmlAttribute);`

I don't know how to see the code with reflection, so sorry if this not exactly accurate, but I was able to reproduce the issue with my own implicit operator.

If I looked at the stack trace, I might have seen the problem.

I don't know if there's something that can be address for implicit operators, guess not.
